climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+
+LOGGER = logging.getLogger(__name__)
+
+DATA_DIR = str(CONFIG.local_data.save_dir.dir())
+
+FORECAST_DIR = str(Path(DATA_DIR) / 'forecast')
+
+FORECAST_PLOT_DIR = str(Path(FORECAST_DIR) / 'plots')
+
+# defining colormaps
+# The colors are in line the european meteoalarm colors http://www.meteoalarm.info/
+# and with the colors used at MeteoSwiss
+# https://www.natural-hazards.ch/home/dealing-with-natural-hazards/explanation-of-the-danger-levels.html
+# colors for warning levels
+COLORS_WARN = np.array([[204 / 255, 255 / 255, 102 / 255, 1],  # green
+                        [255 / 255, 255 / 255, 0 / 255, 1],  # yellow
+                        [255 / 255, 153 / 255, 0 / 255, 1],  # orange
+                        [255 / 255, 0 / 255, 0 / 255, 1],  # red
+                        [128 / 255, 0 / 255, 0 / 255, 1],  # dark red
+                        ])
+# colors for warning probabilities
+warnprob_colors = np.array([[255 / 255, 255 / 255, 255 / 255, 1],  # white
+                            [215 / 255, 227 / 255, 238 / 255, 1],  # lightest blue
+                            [181 / 255, 202 / 255, 255 / 255, 1],  # ligthblue
+                            [127 / 255, 151 / 255, 255 / 255, 1],  # blue
+                            [171 / 255, 207 / 255, 99 / 255, 1],  # green
+                            [232 / 255, 245 / 255, 158 / 255, 1],  # light yellow?
+                            [255 / 255, 250 / 255, 20 / 255, 1],  # yellow
+                            [255 / 255, 209 / 255, 33 / 255, 1],  # light orange
+                            [255 / 255, 163 / 255, 10 / 255, 1],  # orange
+                            [255 / 255, 76 / 255, 0 / 255, 1],  # red
+                            ])
+warnprob_colors_extended = np.repeat(warnprob_colors, 10, axis=0)
+CMAP_WARNPROB = ListedColormap(warnprob_colors_extended)
+# colors for impact forecast
+color_map_pre = plt.get_cmap('plasma', 90)
+impact_colors = color_map_pre(np.linspace(0, 1, 90))
+white_extended = np.repeat([[255 / 255, 255 / 255, 255 / 255, 1]], 10, axis=0)
+impact_colors_extended = np.append(white_extended, impact_colors, axis=0)
+CMAP_IMPACT = ListedColormap(impact_colors_extended)
+
+
+class Forecast():
+    """Forecast definition. Compute an impact forecast with predefined hazard
+    originating from a forecast (like numerical weather prediction models),
+    exposure and impact. Use the calc() method to calculate a forecasted
+    impact. Then use the plotting methods to illustrate the forecasted impacts.
+    As the class is relatively new, there might be future changes to the attributes,
+    the methods, and the parameters used to call the methods
+
+    Attributes
+    ----------
+    run_datetime: list
+        List of datetime.datetime,
+        initialization time of the forecast model run used to create the Hazard
+    event_date: datetime.datetime
+        Date on which the Hazard event takes place
+    hazard: list
```

Comment:
Probably it would be good to rename this as `hazard_list`. Otherwise the code is a bit confusing.